proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,183 @@
+ any package that was staged but never published within the specified / pre-configured time period. As a maximum timeslot
+ a default one month period can be considered. However, to reduce the cost of maintaining these packages,
+ the `nuget stage` command can also accept an optional `--lifespan [number of days]` parameter,
+ which can indicate a shorter time period, thus giving NuGet ability to remove those packages earlier than the maximum period.
+
+Note, that this opens up a new scenario, where a group has been created with a package 29 days ago, and then a new package is staged to the same group.
+In two days from that time, the package author tries to publish the staged group. At this point, the publish command will fail,
+indicating that the group has been modified, and showing the package that has been deleted by NuGet out of the staging due to the expiration.
+The user can still continue publishing using the `--force` option, which will result in the non-expired/ removed staged packages to be published.
+Alternatively, the user can restage the expired package and call the publish command again.
+
+#### Expanding groups
+It should be possible for a group to expand over time. This is currently a real scenario, where .NET GA SDK ships a set of packages,
+ and then follow-up SDK updates ship updated packages which still belong to the same .NET 8 package group.
+ Here how this will play out with the tooling support described above:
+1. During GA release, the release team will stage the set of RTM packages for the release. (nuget stage --group-id "net8.0" --package-id <package-id>)
+2. Then on the day of the release, all the staged packages will be published using `nuget publish --group-id "net8.0"` command.
+   At this point, the stage for the `net8.0` group will be empty.
+3. Later, as new builds are being prepared for a patch release, a new set of packages will be staged, to be later published to the same group.
+As there can be multiple candidate builds for a release, there is a need for functionality to discard a stage for a given group `nuget stage --discard --group-id "net8.0"`.
+This will remove all the staged but not published packages from the net8.0 group.
+4. A new set of packages will be staged for the 8.0 group, untile the final build is known. At this point, there is only one unified set of related packages is staged.
+5. On the release date, the release team will call `nuget publish --group-id "net8.0"` again, and only the staged packages will be published, resulting in an expanded set of published packages in the net8.0 group.
+6. Some time in the future, when the release is already out of support, somebody from the .NET team will call
+ `nuget deprecate --group-id "net8.0"` and all the packages which have ever been published to that group will be deprecated.
+ Those packages, which has already been deprecated because of whatever reason, will not be altered.
+
+#### Asynchronous behavior
```

Comment:
I know I haven't called out how these async operations will be tracked. I'll work on this and include details about this too.